homeassistant/components/arcam_fmj/config_flow.py

```diff
@@ -1,12 +1,63 @@
 """Config flow to configure the Arcam FMJ component."""
-from operator import itemgetter
+import asyncio
+import logging
+from urllib.parse import urlparse
+
+import aiohttp
+from defusedxml import ElementTree
+import voluptuous as vol
 
 from homeassistant import config_entries
+from homeassistant.components.ssdp import ATTR_SSDP_LOCATION, ATTR_UPNP_UDN
 from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.data_entry_flow import AbortFlow
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DEFAULT_NAME, DEFAULT_PORT, DOMAIN, DOMAIN_DATA_ENTRIES
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _strip_uuid_prefix(data):
+    return data[5:].split("-")[4]
+
+
+def _log_exception(msg, *args):
+    """Log an error and turn on traceback if debug is on."""
+    _LOGGER.error(msg, *args, exc_info=_LOGGER.getEffectiveLevel() == logging.DEBUG)
+
+
+async def _get_uniqueid_from_device_description(hass, url):
+    """Retrieve and extract unique id from url."""
+    try:
+        session = async_get_clientsession(hass)
+        async with session.get(url) as req:
```

Comment:
This should be moved to the library.

Reply:
Potentially. But library have no need of unique ID. It is also something that should be part of homeassistants ssdp client. There are other integrations doing the same type of thing.

Reply:
Device interface code should be in a 3rd party library not in home assistant. If we decide to make some general helper that's a different question and PR.